src/pages/Scholarships.jsx

```diff
@@ -1,62 +1,136 @@
 import React, { useState, useEffect } from 'react';
+import { useHistory, useLocation } from 'react-router-dom';
 import {
+  Button,
+  Box,
   CircularProgress,
   Container,
   makeStyles,
   Typography,
+  Grid,
+  useMediaQuery,
 } from '@material-ui/core';
-import Button from '@material-ui/core/Button';
+import queryString from 'query-string';
 import Scholarships from '../models/Scholarships';
 import ScholarshipList from '../components/ScholarshipList';
 import FilterBar from '../components/FilterBar';
+import ScholarshipDetailCard from '../components/ScholarshipDetailCard';
 
 const useStyles = makeStyles((theme) => ({
   progress: {
     display: 'block',
     margin: 'auto',
   },
+  resultsArea: {
+    display: 'flex',
+    flexDirection: 'column',
+    flexGrow: 1,
+    justifyContent: 'center',
+    alignItems: 'center',
+    minHeight: '20vh',
+  },
   loadMoreButton: {
     margin: theme.spacing(3, 0),
   },
 }));
 
 function ScholarshipsPage() {
+  const classes = useStyles();
   const [scholarships, setScholarships] = useState([]);
   const [loading, setLoading] = useState(true);
   const [error, setError] = useState();
   const [sortField, setSortField] = useState('deadline');
   const [sortDir, setSortDir] = useState('asc');
 
   useEffect(() => {
-    // TODO: Create cancellable promises
+    let mounted = true;
     Scholarships.list({ sortField, sortDir })
-      .then(setScholarships)
-      .catch(setError)
-      .finally(() => setLoading(false));
+      .then((results) => {
+        if (mounted) setScholarships(results);
+      })
+      .catch((e) => {
+        if (mounted) setError(e);
+      })
+      .finally(() => {
+        if (mounted) setLoading(false);
+      });
+    return () => {
+      mounted = false;
+    };
   }, [sortDir, sortField]);
 
-  const classes = useStyles();
+  const history = useHistory();
+  const location = useLocation();
+  const params = queryString.parse(location.search, { parseNumbers: true });
+
+  // Parse whether an item has been selected
+  let selectedIndex = params.selectedIndex ?? undefined;
```

Comment:
TODO: Use scholarship ID instead. Index might be a bit fragile if the position of the scholarship shifts due to scholarships expiring or new scholarships pushing it down.